modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/NGramTokenizerFactoryTests.java

```diff
@@ -152,6 +153,34 @@ public void testBackwardsCompatibilityEdgeNgramTokenFilter() throws Exception {
     }
 
 
+    /*`
+    * test that throws an error when trying to get a NGramTokenizer where difference between max_gram and min_gram
+    * is greater than the allowed value of max_ngram_diff
+     */
+    public void testMaxNGramDiffException() throws Exception{
+        final Index index = new Index("test", "_na_");
+        final String name = "ngr";
+        final Settings indexSettings = newAnalysisSettingsBuilder().build();
+        IndexSettings indexProperties = IndexSettingsModule.newIndexSettings(index, indexSettings);
+
+        int maxAllowedNgramDiff = indexProperties.getMaxNgramDiff();
+        int ngramDiff = maxAllowedNgramDiff + 1;
+        int min_gram = 2;
+        int max_gram = min_gram + ngramDiff;
+
+        final Settings settings = newAnalysisSettingsBuilder().put("min_gram", min_gram).put("max_gram", max_gram).build();
+        try {
+            new NGramTokenizerFactory(indexProperties, null, name, settings).create();
+            fail();
```

Comment:
Maybe use `LuceneTestCase#expectThrows`

Reply:
I was trying to use `LuceneTestCase#expectThrows`, but it requires that  a constructor`public NGramTokenizerFactory(..) ` throw a checked exception.  We can add this to the constructor, but then we need to modify all other classed that use it to catch this exception

Reply:
It works fine with a runtime exception, you don't need to change the constructor:
```
IllegalArgumentException exc = expectThrows(IllegalArgumentException.class, () ->
                new NGramTokenizerFactory(indexProperties, null, name, settings).create());
```
.. and then you can check the message of the exception.

Reply:
thanks Jim, will try this